src/backend/aspen/api/conftest.py

```diff
@@ -0,0 +1,122 @@
+import random
+import string
+from dataclasses import dataclass
+from functools import partial
+from typing import AsyncGenerator
+
+import pytest
+from fastapi import Depends, FastAPI, Request
+from httpx import AsyncClient
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from aspen.api.auth import get_auth_user, setup_userinfo
+from aspen.api.deps import get_db
+from aspen.api.main import get_app
+from aspen.database import connection as aspen_connection
+from aspen.database import schema
+from aspen.database.connection import init_async_db
+
+USERNAME = "user_rw"
+PASSWORD = "password_rw"
+INITIAL_DATABASE = "postgres"
+
+pytestmark = pytest.mark.anyio
+
+
+@dataclass
+class AsyncPostgresDatabase:
+    database_name: str
+    port: int
+
+    def as_uri(self) -> str:
+        return f"postgresql+asyncpg://{USERNAME}:{PASSWORD}@database:{self.port}/{self.database_name}"
+
+
+@pytest.fixture()
```

Comment:
I could be misunderstanding something, but it seems to me like creating a new DB would be a pretty heavy operation and one we would want to avoid doing for every test func, but doesn't the pytest fixture default to a function scope? https://docs.pytest.org/en/6.2.x/fixture.html#fixture-scopes

Is the intention to create a test DB for every test func, or would it be better to do it less frequently than that? I also could be wrong about my understanding of when the fixture is happening here, I've definitely beaten my head against pytest fixtures in the past.

Reply:
For the moment I'm just mimic'ing our existing test implementation for now, which creates db's with a function scope, though I think it's important to be able to share a db between tests -- we need to make sure our endpoints are using DB isolation properly.